### PR TITLE
Introduce hook to manage error messages for modules

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.6.x
| Description?  | Add a hook that allow to manage the mapping in controllers for error messages
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes https://github.com/PrestaShop/PrestaShop/issues/13490 last issue: "How to handle errors happening inside module (translation + display to user)"
| How to test?  | Incoming

### Example of usage

In a module that extends the Customer Form, so it's hooked on `hookActionAfterUpdateCustomerFormHandler`

If an error is found in hook `hookActionAfterUpdateCustomerFormHandler`, we can raise a `CustomerException` (because it's related CustomerController which catches CustomerException only)
```
    public function hookActionAfterUpdateCustomerFormHandler(array $params)
    {
        throw new CustomerException(
            'This is a beautiful error message'
        );
    }
```
And we make sure this Customer Exception has proper error message using the new hook:
```
    public function hookActionGetErrorMessagesInController(array $params)
    {
        $messages = $params['messages'];
        $exception = $params['exception'];
        $controllerName = $params['controller_name'];

        // filter by controller name as hook is generic
        if ($controllerName !== 'PrestaShopBundle\Controller\Admin\Sell\Customer\CustomerController') {
            return;
        }

        $messages[CustomerException::class] = 'Custom error message : '.$exception->getMessage();
        $params['messages'] = $messages;
    }
```

Final result:
Capture d’écran 2019-06-13 à 17 19 25

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/14200)

### CODE REVIEW

**Review by PierreRambaud — DISMISSED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]
Thanks @matks

File: src/PrestaShopBundle/Controller/Admin/FrameworkBundleAdminController.php [hunk 1]
```diff
@@ -465,6 +465,12 @@ protected function getErrorMessageForException(Exception $e, array $messages)
         $exceptionType = get_class($e);
         $exceptionCode = $e->getCode();
 
+        $this->get('prestashop.core.hook.dispatcher')->dispatchWithParameters('actionGetErrorMessagesInController', [
```
Comment (sarjon) [inline src/PrestaShopBundle/Controller/Admin/FrameworkBundleAdminController.php:468]: I'd change it to `actionSetErrorMessagesInController` or `actionAddErrorMessagesInController` as this hook is meant for extending messeges and not getting them, wdyt?

**Review by sarjon — COMMENTED** [review]
The most simple solution. :+1:

File: src/PrestaShopBundle/Controller/Admin/FrameworkBundleAdminController.php (hunk 1 above)
Comment (matks) [reply src/PrestaShopBundle/Controller/Admin/FrameworkBundleAdminController.php:468]: We have another idea - again 😅. Pablo is worried by another usecase.
We'll give it a try tomorrow morning.

**Comment by matks** [comment]
Closed in favor of https://github.com/PrestaShop/PrestaShop/pull/14206

**Pull request closed without merging**